src/core/templates/show-people.js

```diff
@@ -62,113 +81,217 @@ const orcidIcon = () => html`<svg
+    return false;
+  }
+
+  if (person.url && person.mailto) {
+    const msg = `Person object at index ${index} has both "url" and "mailto" property.`;
+    showWarning(msg, name, {
+      hint: `Please chose either "url" or "mailto" ("url" is preferred). ${hint}`,
+    });
+  }
+
+  if (person.companyURL && !person.company) {
+    const msg = `Person object at index ${index} has a "companyURL" member but no "company" member.`;
+    showWarning(msg, name, { hint: `Please add a "company" member. ${hint}` });
+  }
+  return true;
+}
+
+/**
+ *
+ * @param {PersonExtras} extras
+ * @param {string} hint
+ */
+function validateExtras(extras, hint) {
+  if (!Array.isArray(extras)) {
+    showError(`A person's "extras" member must be an array.`, name, { hint });
+    return false;
+  }
+  return extras.every((extra, index) => {
+    switch (true) {
+      case typeof extra !== "object":
+        showError(`"extra" index ${index} is not an object.`, name, {
+          hint,
+        });
+        return false;
+      case !extra.hasOwnProperty("name"):
+        showError(
+          `\`PersonExtra\` object at index ${index} is missing required "name" member.`,
+          name,
+          { hint }
         );
-      } else {
-        contents.push(html` (${company}) `);
-      }
-    }
-    if (p.note) contents.push(document.createTextNode(` (${p.note})`));
-    if (p.extras) {
-      const results = p.extras
-        // Remove empty names
-        .filter(extra => extra.name && extra.name.trim())
-        // Convert to HTML
-        .map(getExtra);
-      for (const result of results) {
-        contents.push(document.createTextNode(", "), result);
-      }
-    }
-    if (p.retiredDate) {
-      const retiredDate = new Date(p.retiredDate);
-      const isValidDate = retiredDate.toString() !== "Invalid Date";
-      const timeElem = document.createElement("time");
-      timeElem.textContent = isValidDate
-        ? humanDate(retiredDate)
-        : "Invalid Date"; // todo: Localise invalid date
-      if (!isValidDate) {
-        const msg = "The date is invalid. The expected format is YYYY-MM-DD.";
-        const title = "Invalid date";
-        showError(msg, name, { title, elements: [timeElem] });
-      }
-      timeElem.dateTime = toShortIsoDate(retiredDate);
-      contents.push(html` - ${l10n.until(timeElem)} `);
+        return false;
+      case typeof extra.name === "string" && extra.name.trim() === "":
+        showError(
+          `\`PersonExtra\` object at index ${index} "name" can't be empty.`,
+          name,
+          { hint }
+        );
+        return false;
     }
+    return true;
+  });
+}
+
+/**
+ * @param {string} member
+ * @param {string} rawDate
+ * @param {number} index
+ * @param {string} hint
+ * @returns
+ */
+function validateDateMember(member, rawDate, index, hint) {
+  const date = new Date(rawDate);
+  const isValidDate = date.toString() !== "Invalid Date";
```

Comment:
Maybe also add a check for `YYYY-MM-DD` format using regex: `/\d{4}-\d{2}-\d{2}/`

Reply:
Good idea.